include/class.thread.php

```diff
@@ -803,10 +803,26 @@ function getFiles() {
 
     function saveEmailInfo($vars) {
 
-        if(!$vars || !$vars['mid'])
+        if (!$vars || !isset($vars['mid']))
             return 0;
 
-        $this->ht['email_mid'] = $vars['mid'];
+        // If the email contains more than one Message-ID header, use the
+        // last non-empty value
+        if (isset($vars['mid']) && is_array($vars['mid'])) {
+            foreach ($vars['mid'] as $M) {
```

Comment:
I think that is the most elegant solution. It will result in an empty array which equates with FALSE